spec/src/main/resources/META-INF/smithy/bsp/extensions/rust.smithy

```diff
@@ -0,0 +1,197 @@
+}
+
+structure RustWorkspaceParams {
+    @required
+    targets: BuildTargetIdentifiers
+}
+
+structure RustWorkspaceResult {
+    @required
+    packages: RustPackages
+    @required
+    rawDependencies: RustRawDependencies
+    @required
+    dependencies: RustDependencies
+}
+
+list RustPackages {
+    member: RustPackage
+}
+
+structure RustPackage {
+    @required
+    id: String
+    version: String
+    origin: String
+    edition: String
+    source: String
+    @required
+    targets: RustTargets
+    @required
+    allTargets: RustTargets
+    @required
+    features: RustFeatures
+    @required
+    enabledFeatures: Strings
+    cfgOptions: RustCfgOptions
+    @required
+    env: RustEnvDatas
+    outDirUrl: String
+    procMacroArtifact: RustProcMacroArtifact
+}
+
+list RustTargets {
+    member: RustTarget
+}
+
+@dataKind(kind: "rust", extends: BuildTargetData)
+structure RustTarget {
+    @required
+    name: String
+    @required
+    crateRootUrl: String
+    @required
+    packageRootUrl: String
+    @required
+    kind: String
+    edition: String
+    doctest: Boolean
+    requiredFeatures: Strings
+}
+
+list RustFeatures {
+    member: RustFeature
+}
+
+structure RustFeature {
+    @required
+    name: String
+    deps: Strings
+}
+
+list Strings {
+    member: String
+}
+
+structure RustCfgOptions {
+    keyValueOptions: KeyValueOptions
+    nameOptions: Strings
+}
+
+map KeyValueOptions {
+    key: String
+    value: Strings
+}
+
+map RustEnvDatas {
+    key: String
+    value: String
+}
+
+structure RustProcMacroArtifact {
+    path: String
+    // TODO: we don't need hash. It is calculated by IntelliJ-Rust
+}
+
+map RustRawDependencies {
+    key: String // PackageId
+    value: RustRawDependency
+}
+
+structure RustRawDependency {
+    @required
+    name: String
+    rename: String
+    kind: String
+    target: String
+    optional: Boolean
+    uses_default_features: Boolean
+    @required
+    features: Strings
+}
+
+map RustDependencies {
+    key: String // Source
+    value: RustDependency
+}
+
+structure RustDependency {
+    @required
+    target: String
+    name: String
+    depKinds: DepKinds
+}
+
+list DepKinds {
+    member: DepKind
+}
+
+structure DepKind {
+    @required
+    kind: String
+    target: String
+}
+
+@jsonRequest("buildTarget/rustToolchain")
+operation RustToolchain {
+    input: RustToolchainParams
+    output: RustToolchainResult
+}
+
+structure RustToolchainParams {
+    @required
+    targets: BuildTargetIdentifiers
+}
+
+structure RustToolchainResult {
+    @required
+    items: RustToolchainsItems
+}
+
+list RustToolchainsItems {
+    member: RustToolchainsItem
+}
+
+structure RustToolchainsItem {
+    rustStdLib: RustcInfo
+    @required
+    cargoBinPath: String
+    @required
+    procMacroSrvPath: String
+}
+
+structure RustcInfo {
+    @required
+    sysroot: String
+    @required
+    srcSysroot: String
+    @required
+    version: String
+    @required
+    host: String
```

Comment:
what is it and what is it for?

Reply:
can you put it in the description? Is it obtained from cargo metadata?

Reply:
It is obtained from `rustc`

Reply:
I checked once again, and it's more important usage is checking if given toolchain is supported with this specified host. I will put it in description